### PR TITLE
Geo/11003-ManageMultipleMovingPointLoads

### LINKED ISSUE #11003: [GeoMechanicsApplication][STEM] Create process managing a collection of moving_loads
Create process managing a collection of moving_loads
- initialization, operations must be consistent
- multiple moving loads based on a configuration translating to an offset

This action is needed for issue #10947 as in the issue multiple moving point loads are spawned to simulate the contact points of a moving vehicle.

### PR DESCRIPTION
**Pull request by mcgicjn2**
**📝 Description**
Create process managing a collection of moving_loads

- initialization, operations must be consistent
- multiple moving loads based on a configuration translating to an offset

This will be a PR depends on the structuralmechanics application until PR #10997 is merged

Closes #11003 

**🆕 Changelog**
Added process to set and manage a collection multiple moving point loads
- clones mutliple moving_line_conditions into compute model part
- removes associated original moving line conditions from compute model part
- manages list of multiple set_moving_line_load processes + calls
+ some associated tests

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        KRATOS_ERROR_IF(!is_all_string && !is_all_number) << "'load' has to be a vector of numbers, or an array with strings" << std::endl;
+
+
+        int count = 0;
+    	for (double offset : mParameters["configuration"].GetVector())
+        {
+
+            auto parameters_moving_load = mParameters.Clone();
+
+    		count++;
```
Comment (huhaas) [inline applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp]: Indents are not consistent due to tab usage. In multiple files.

File: applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+         ///@}
+
+    private:
+        ///@name Member Variables
+        ///@{
+
+	        ModelPart& mrModelPart;
+	        Parameters mParameters;
+	        std::vector<SetMovingLoadProcess> mMovingPointLoadsProcesses;
```
Comment (huhaas) [inline applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.h]: Maybe it is better to use the Kratos unique pointer construction here. Currently the push_back in cpp line 80 creates unwanted copies.

File: applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    		auto r_moving_point_process = SetMovingLoadProcess(new_cloned_model_part, parameters_moving_load);
+            mMovingPointLoadsProcesses.push_back(r_moving_point_process);
+           
+        }
+
+        RemoveClonedConditions();
+
+    }
+
+    ModelPart& SetMultipleMovingLoadsProcess::CloneMovingConditionInComputeModelPart(std::string NewBodyPartName)
```
Comment (huhaas) [inline applications/GeoMechanicsApplication/custom_processes/set_multiple_moving_loads.cpp]: Argument can be const&

File: applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py [hunk 4]
```diff
@@ -0,0 +1,1886 @@
+import KratosMultiphysics
+
+import KratosMultiphysics.GeoMechanicsApplication as GMA
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+class TestSetMultipleMovingLoadsProcess(KratosUnittest.TestCase):
```
Comment (huhaas) [inline applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py:6]: I have some suggestions to improve this class. Let's discuss this in the office. Also SonarCloud does not agree with this change.

**Review by huhaas — CHANGES_REQUESTED** [review]
Please update

File: applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py [hunk 5]
```diff
@@ -26,18 +26,20 @@ def setUp(self):
...
-        self.assertAlmostEqual(rhs[1], expected_res[1])
-        self.assertAlmostEqual(rhs[2], expected_res[2])
-        self.assertAlmostEqual(rhs[3], expected_res[3])
-
+        
+        if tols is None:
+            tols = [None] * len(rhs)
+        
+        for rhs_val, expected_val, tols in zip(rhs, expected_res, tols):
+            self.assertAlmostEqual(rhs_val, expected_val, tols)
```
Comment (huhaas) [inline applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py]:
```suggestion
        for rhs_val, expected_val, tol in zip(rhs, expected_res, tols):
            self.assertAlmostEqual(rhs_val, expected_val, tol)
```

**Comment by aronnoordam** [comment]
LGTM

**Review by aronnoordam — DISMISSED** [review]

**Review by avdg81 — APPROVED** [review]
I have made several minor coding improvements (see my last commit). It looks good to me now, Jon. Thanks for the hard work that you've put into this!

**Review by huhaas — APPROVED** [review]

**Pull request merged**